config/schema.json

```diff
@@ -62,7 +62,8 @@
       "properties": {
         "name": {
           "description": "The name the service will be registered as in Consul. Defaults to the Task family name if empty or null.",
-          "type": ["string", "null"]
+          "type": ["string", "null"],
+          "pattern": "(^$)|(^[a-z0-9]([a-z0-9-_]*[a-z0-9])?$)"
```

Comment:
This is empty string `(^$)` or the [service identity regex](https://github.com/hashicorp/consul/blob/93b164aac37f83863707f05284b868dc8ea57719/acl/validation.go#L11) from Consul.